### DISCUSSION TITLE
Do Projects Need Better Maps?

### DISCUSSION THREAD

**Discussion by mo-nathan** [MushroomObserver/mushroom-observer#1900] [General]
When I talked with Steven Russell at NAMA 2023, I learned that he really likes maps.  Currently we allow users to map observations, but it's kind of buried.  Should we make this more prominent?  Maybe even add it to project show page?  Seems like some sort of heat mappy thing would be best, but we could certainly go with our current functionality.  A potentially cheaper option would just be a map of any project constraint.  Using the "First Fantastic Fungal Friends Foray Festival!" as an example, that would currently be something like, https://mushroomobserver.org/locations/358, which I don't find very satisfying.  The map of observations is https://mushroomobserver.org/observations/map?q=1pcjX (but I'm not sure how long queries persist these days, so I'll also attach a screen grab) which is more satisfying.
<img width="308" alt="Screenshot 2024-02-11 at 7 30 55 AM" src="https://github.com/MushroomObserver/mushroom-observer/assets/3211730/b86aa79a-69c2-4f05-8051-1e53b201122b">

**Comment by nimmolo** [comment]
Just to have it recorded here in the discussion: 
- to use Google's heat maps or other clustering functions (and not have to code our own), we will need to provide lat/lng points for our many observations that lack them
- we'll need to consider ensuring that the lat/lngs are unique enough that the points do not coincide exactly when zoomed in, if we want them to be clickable. My idea was to fudge-randomize the center points of named Locations to some extent and give them a different color indicating the placement is approximate, but it's also possible to have "irreducible clusters", in other words a big "15" on a cluster dot that does not resolve to separate points no matter how closely you zoom. Because of the prevalence of "no lat/lng obs" in our db, to me that may produce very counterintuitive maps in many cases.

**Comment by mo-nathan** [comment]
Between those choices I think I'd rather have the clusters than fudge stuff.  It's the data we have.  Is there a concept of error that we should be supporting?  I'm wondering if we need to have a bigger discussion around locations.  At the moment I'm feeling like if we get a lat/long that's all we should need.  I don't know what the current situation is with reverse geocoding, but that sure seems like what we should be doing if we can.  If we could just get rid of the location selection stuff from the app that would be sweet.
